system/jlib/jlz4.cpp

```diff
@@ -126,11 +127,50 @@ class CLZ4Compressor final : public CFcmpCompressor
         return outlen;
     }
 
+    virtual bool supportsBlockCompression() const override { return true; }
+    virtual bool supportsIncrementalCompression() const override { return false; }
+
+    virtual size32_t compressBlock(size32_t destSize, void * dest, size32_t srcSize, const void * src) override
+    {
+        if (destSize <= 3 * sizeof(size32_t))
+            return 0;
+
+        //Format is total-uncompressed-size, (compressed size, compressedData)* (uncompressed size, data)
```

Comment:
Slightly confusing comment. The "uncompressed size" at the end refers just to the leftover data that doesn't form any part of the compressed data, where the one at the start refers to the entire input data. Perhaps reword to make that clear?

Reply:
reworded